lib/cylc/task_events_mgr.py

```diff
@@ -993,28 +993,31 @@ def _reset_job_timers(self, itask):
             timeref = itask.summary['started_time']
             timeout_key = 'execution timeout'
             timeout = self._get_events_conf(itask, timeout_key)
+            # delays - All polling times after start
+            # timeout - Total time limit including all polling:
             delays = list(self.get_host_conf(
                 itask, 'execution polling intervals', skey='job',
                 default=[900]))  # Default 15 minute intervals
             if itask.summary[self.KEY_EXECUTE_TIME_LIMIT]:
                 time_limit = itask.summary[self.KEY_EXECUTE_TIME_LIMIT]
+
+                # Get execution time limit polling intervals or set to default:
                 try:
                     host_conf = self.get_host_conf(itask, 'batch systems')
                     batch_sys_conf = host_conf[itask.summary['batch_sys_name']]
                 except (TypeError, KeyError):
                     batch_sys_conf = {}
                 time_limit_delays = batch_sys_conf.get(
                     'execution time limit polling intervals', [60, 120, 420])
-                timeout = time_limit + sum(time_limit_delays)
-                # Remove excessive polling before time limit
-                while sum(delays) > time_limit:
-                    del delays[-1]
-                # But fill up the gap before time limit
-                if delays:
-                    size = int((time_limit - sum(delays)) / delays[-1])
-                    delays.extend([delays[-1]] * size)
-                time_limit_delays[0] += time_limit - sum(delays)
-                delays += time_limit_delays
+
+                # Total timeout after adding execution time limit polling
+                # intervals:
+                timeout = (time_limit + sum(time_limit_delays))
```

Comment:
This line seems to be missing from the Cylc 8 branch?

Reply:
It was removed here: https://github.com/cylc/cylc-flow/pull/4863 - @dpmatthews  is it sane to backport this change?

Reply:
Actually, this bit is a change in behaviour rather than a bug fix. I suggest we leave it unchanged in Cylc 7.